File: nim/words/tests/test_numerical.nim
import unittest
import numerical

suite "Iterative Suite":
    test "fibonacci":
        assert fibonacci(-3) == -1
        assert fibonacci(10) == 55
        assert fibonacci(15) == 610
    
    test "factorial":
        assert factorial(-3) == -1
        assert factorial(5) == 120
        assert factorial(10) == 3628800

    test "greatest common divisor":
        assert gcd(48, 18) == 6
        assert gcd(100, 25) == 25
        assert gcd(17, 13) == 1

    test "lowest common multiple":
        assert lcm(6, 4) == 12
        assert lcm(21, 6) == 42
        assert lcm(7, 3) == 21

File: ocaml/words/test/test_numerical.ml
open OUnit2
module Numerical = Liby.Numerical

let suite_numerical = "Numerical Suite" >:::
  [
    "test_fibonacci" >:: (fun _ ->
      assert_equal (-1) (Numerical.fibonacci (-3));
      assert_equal 55 (Numerical.fibonacci 10);
      assert_equal 610 (Numerical.fibonacci 15);
    );

    "test_factorial" >:: (fun _ ->
      assert_equal (-1) (Numerical.factorial (-3));
      assert_equal 120 (Numerical.factorial 5);
      assert_equal 3628800 (Numerical.factorial 10);
    );

    "test_gcd" >:: (fun _ ->
      assert_equal 6 (Numerical.gcd 48 18);
      assert_equal 25 (Numerical.gcd 100 25);
      assert_equal 1 (Numerical.gcd 17 13);
    );

    "test_lcm" >:: (fun _ ->
      assert_equal 12 (Numerical.lcm 4 6);
      assert_equal 42 (Numerical.lcm 21 6);
      assert_equal 21 (Numerical.lcm 7 3);
    );
  ]

let () =
  run_test_tt_main suite_numerical